shared/actions/chat2/index.tsx

```diff
@@ -3588,6 +3588,15 @@ const maybeChangeChatSelection = (action: RouteTreeGen.OnNavChangedPayload, logg
   const {prev, next} = action.payload
   const p = prev[prev.length - 1]
   const n = next[next.length - 1]
+
+  const wasModal = prev[1]?.routeName !== 'Main'
+  const isModal = next[1]?.routeName !== 'Main'
+
+  // ignore if changes involve a modal
+  if (wasModal || isModal) {
```

Comment:
if you're on a modal on the people tab that calls `switchTab` to chat, will this bail on reselecting?

Reply:
its a good question. it should be impossible as there's a series of navactions that are driven by the navigateToThread so the modal changes happen early. We'll eventually end with a setParams action to plumb the convo id to the route.
to test i went to a profile, clicked ..., send lumens, hover over the name, click chat, and it works correctly

Reply:
Ok cool, thanks for checking!